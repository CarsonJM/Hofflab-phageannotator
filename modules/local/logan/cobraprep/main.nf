process LOGAN_DOWNLOAD {
    tag "${meta.id}"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/awscli:1.32.101--263008d47506de8c' :
        'community.wave.seqera.io/library/awscli:1.32.101--bca0ff06f80a537e' }"

    input:
    tuple val(meta), path(accessions)

    output:
    tuple val(meta), path("${prefix}.fa.zst")   , emit: fasta
    path "versions.yml"                         , emit: versions

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir tmp
    cat ${accessions} | xargs -I{} -n 1 -P ${task.cpus} aws s3 cp s3://logan-pub/c/{}/{}.contigs.fa.zst tmp --no-sign-request
    cat tmp/*.fa.zst > ${prefix}.fa.zst
    rm -rf tmp

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        awscli: \$( aws --version | sed 's/aws-cli\\///; s/ Python.*//' )
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.fa.zst

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        awscli: \$( aws --version | sed 's/aws-cli\\///; s/ Python.*//' )
    END_VERSIONS
    """
}
