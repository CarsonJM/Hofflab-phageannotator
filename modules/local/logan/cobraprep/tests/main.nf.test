nextflow_process {

    name "Test process: LOGAN_DOWNLOAD"
    script "../main.nf"
    process "LOGAN_DOWNLOAD"


    test("logan_accessions.txt") {

        when {
            params {
                logan_accession_files   = "${projectDir}/assets/test-datasets/logan/logan_group*_accessions.txt"
            }
            process {
                """
                input[0] = Channel.fromFilePairs( params.logan_accession_files, size: 1 )
                    .map { meta, accession_file ->
                        [ [ id:"logan_acc_" + meta ], accession_file ]
                    }
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("logan_accessions.txt - stub") {

        options "-stub"

        when {
            params {
                logan_accession_files   = "${projectDir}/assets/test-datasets/logan/logan_group*_accessions.txt"
            }
            process {
                """
                input[0] = Channel.fromFilePairs( params.logan_accession_files, size: 1 )
                    .map { meta, accession_file ->
                        [ [ id:"logan_acc_" + meta ], accession_file ]
                    }
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
