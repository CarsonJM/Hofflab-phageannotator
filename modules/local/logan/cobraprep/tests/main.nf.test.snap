{
    "logan_accessions.txt": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "logan_acc_logan_group1"
                        },
                        "logan_acc_logan_group1.fa.zst:md5,1ef077d5affed09fad49c6fa1cb1870b"
                    ],
                    [
                        {
                            "id": "logan_acc_logan_group2"
                        },
                        "logan_acc_logan_group2.fa.zst:md5,f76efa24154513098f3c71bcb8d3f51e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,51cf3a68e8516af227452d4c023ece2d",
                    "versions.yml:md5,51cf3a68e8516af227452d4c023ece2d"
                ],
                "fasta": [
                    [
                        {
                            "id": "logan_acc_logan_group1"
                        },
                        "logan_acc_logan_group1.fa.zst:md5,1ef077d5affed09fad49c6fa1cb1870b"
                    ],
                    [
                        {
                            "id": "logan_acc_logan_group2"
                        },
                        "logan_acc_logan_group2.fa.zst:md5,f76efa24154513098f3c71bcb8d3f51e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,51cf3a68e8516af227452d4c023ece2d",
                    "versions.yml:md5,51cf3a68e8516af227452d4c023ece2d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-06-12T14:15:48.382013125"
    },
    "logan_accessions.txt - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "logan_acc_logan_group1"
                        },
                        "logan_acc_logan_group1.fa.zst:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "logan_acc_logan_group2"
                        },
                        "logan_acc_logan_group2.fa.zst:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,51cf3a68e8516af227452d4c023ece2d",
                    "versions.yml:md5,51cf3a68e8516af227452d4c023ece2d"
                ],
                "fasta": [
                    [
                        {
                            "id": "logan_acc_logan_group1"
                        },
                        "logan_acc_logan_group1.fa.zst:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "logan_acc_logan_group2"
                        },
                        "logan_acc_logan_group2.fa.zst:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,51cf3a68e8516af227452d4c023ece2d",
                    "versions.yml:md5,51cf3a68e8516af227452d4c023ece2d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-06-12T14:16:01.412050421"
    }
}